include/forcing/GridDataSelector.hpp

```diff
@@ -0,0 +1,300 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#include <boost/core/span.hpp>
+
+#include <geojson/JSONGeometry.hpp>
+
+struct Cell {
+    std::uint64_t x = 0;
+    std::uint64_t y = 0;
+    std::uint64_t z = 0;
+    double    value = NAN;
+
+    friend inline bool operator<(const Cell& a, const Cell& b) {
+        // TODO: Not really meaningful
+        return a.x < b.x || a.y < b.y || a.z < b.z;
+    }
+};
+
+struct Extent {
+    double xmin;
+    double xmax;
+    double ymin;
+    double ymax;
+
+    geojson::polygon_t as_polygon() const noexcept {
+        geojson::polygon_t result;
+        result.outer().reserve(4);
+        result.outer().emplace_back(xmin, ymin);
+        result.outer().emplace_back(xmax, ymin);
+        result.outer().emplace_back(xmax, ymax);
+        result.outer().emplace_back(xmin, ymax);
+        return result;
+    }
+};
+
+struct GridSpecification {
+    //! Total number of rows (aka y)
+    std::uint64_t rows;
+
+    //! Total number of columns (aka x)
+    std::uint64_t columns;
+
+    //! Extent of the grid region (aka min-max corner points)
+    Extent extent;
+};
+
+struct SelectorConfig {
+    //! Initial time for query.
+    //! @todo Refactor to use std::chrono
+    time_t init_time;
+
+    //! Duration for query, in seconds.
+    //! @todo Refactor to use std::chrono
+    long duration_seconds;
+
+    //! Variable to return from query.
+    std::string variable_name;
+
+    //! Units for output variable.
+    std::string variable_units;
+};
+
+struct GridDataSelector {
+
+    //! Cell-based constructor
+    GridDataSelector(SelectorConfig config, boost::span<const Cell> cells) noexcept
```

Comment:
These constructors probably shouldn't be `noexcept`, since they're calling constructors and member functions on containers that will allocate.